<launch>
    <node name="controller" pkg="pid" type="controller" ns="right_wheel" output="screen" >
        <param name="node_name" value="left_wheel_pid" />
        <param name="Kp" value="0.9" />
        <param name="Ki" value="0.1" />
        <param name="Kd" value="0.0" />
        <param name="upper_limit" value="255" />
        <param name="lower_limit" value="-255" />
        <param name="windup_limit" value="10" />
        <param name="max_loop_frequency" value="100.0" />
        <param name="min_loop_frequency" value="100.0" />
        <remap from='setpoint' to='/target_r'/>
        <remap from='state' to='/Encoder_r'/>
        <remap from='control_effort' to='/control_effort_r'/>

    </node>

    <node name="servo_sim_node" pkg="pid" type="plant_sim" ns="right_wheel" output="screen" >
        <param name="plant_order" value="2" />
    </node>

      <node name="controller" pkg="pid" type="controller" ns="left_wheel" output="screen" >
        <param name="node_name" value="left_wheel_pid" />
        <param name="Kp" value="1.05" />
        <param name="Ki" value="0.15" />
        <param name="Kd" value="0.0" />
        <param name="upper_limit" value="255" />
        <param name="lower_limit" value="-255" />
        <param name="windup_limit" value="10" />
        <param name="max_loop_frequency" value="100.0" />
        <param name="min_loop_frequency" value="100.0" />
        <remap from='setpoint' to='/target_l'/>
        <remap from='state' to='/Encoder_l'/>
        <remap from='control_effort' to='/control_effort_l'/>

    </node>

    <node name="servo_sim_node" pkg="pid" type="plant_sim" ns="left_wheel" output="screen" >
        <param name="plant_order" value="2" />
    </node>

    <node pkg = "project_3" type="pid_test.py" name = "pid_test" />

    <node pkg = 'project_3' type='odom.py' name = 'odom'>

    <include file="$(find lidar_ros_delta2a)/launch/view_delta_lidar.launch"/>

    <!-- static Tf brodcast -->
    <node pkg="tf" type="static_transform_publisher" name="map_nav_broadcaster" args="0 0 0 0 0 0 map odom 100"/>
    <node pkg="tf" type="static_transform_publisher" name="laser_nav_broadcaster" args="0 0 0 0 0 0 base_link laser 100"/>

   
</launch>